File: Problems/0035_Search_Insert_Position/Project_CS/Search_Insert_Position.cs
using System;

public class Solution {
    public int SearchInsert(int[] nums, int target)
    {
        int i;
        
        for ( i = 0; i < nums.Length; i++ ) {
            if ( nums[i] < target ) {
                continue;
            }
            else if ( nums[i] >= target ) {
                return i;
            }
        }
        
        return i;
    }

    public int[] str_to_int_array(string s)
    {
        string[] flds = s.Split(',');
        int[] nums = new int[flds.Length];

        if (flds.Length <= 0)
            return nums;

        for (int i = 0; i < nums.Length; ++i)
        {
            nums[i] = int.Parse(flds[i]);
        }

        return nums;
    }

    public string output_int_array(int[] nums)
    {
        if (nums.Length <= 0)
            return "";

        string resultStr = "[" +  nums[0].ToString();
 
        for (int i = 1; i < nums.Length; ++i)
        {
            resultStr += "," + nums[i].ToString();
        }

        return resultStr + "]";
    }

    public void Main(string args)
    {
        string arg_str = args.Replace("[[","").Replace("]]","").Trim();
        string[] flds = arg_str.Split(new string[] {"],["}, StringSplitOptions.None);
        int[] nums = str_to_int_array(flds[0]);
        int target = int.Parse(flds[1]);

        Console.WriteLine("nums = " + output_int_array(nums));
        Console.WriteLine("target = " + target.ToString());

        System.Diagnostics.Stopwatch sw = new System.Diagnostics.Stopwatch();
        sw.Start();

        int result = SearchInsert(nums, target);
        Console.WriteLine("result = " + result.ToString());
        
        sw.Stop();
        Console.WriteLine("Execute time ... " + sw.ElapsedMilliseconds.ToString() + "ms\n");
    }
}

File: Problems/0001_0099/0035_Search_Insert_Position/Project_C/Search_Insert_Position.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>
#include <math.h>
#include <time.h>

#include "../../mylib_C/mylib.h"

/* Function prototype declaration */
int searchInsert(int* nums, int numsSize, int target);
int loop_main(char* arg);

int searchInsert(int* nums, int numsSize, int target)
{
    int i;
    
    for ( i = 0; i < numsSize; i++ ) {
        if ( nums[i] < target ) {
            continue;
        }
        else if ( nums[i] >= target ) {
            return i;
        }
    }
    
    return i;
}

int loop_main(char* arg)
{
    int argv_length = strlen(arg);
    char* flds[2];
    char* str_nums[256];
    int nums[256];

    replace(arg, "[[", "");
    replace(arg, "]]", "");
    replace(arg, "\n", "");
    int flds_length = split(arg, "],[", flds, sizeof(flds)/sizeof(flds[0]));
    int str_numsSize = split(flds[0], ",", str_nums, sizeof(str_nums)/sizeof(str_nums[0]));

    // int nums[] size check.
    if (sizeof(nums)/sizeof(nums[0]) < str_numsSize)
        err_exit("nums[] size error.");

    int numsSize = str_to_int_array(str_nums, nums, str_numsSize);
    int target = atoi(flds[1]);

    clock_t time_start = clock();
    int results = searchInsert(nums, numsSize, target);
    clock_t time_end = clock();

    // result print.
    printf("result = %d\n", results);
    printf("Execute time ... %.0f ms\n\n", 1000*(double)(time_end - time_start)/CLOCKS_PER_SEC);

    // char* str_nums[] free().
    p_char_array_free(str_nums, str_numsSize);

    // char* flds[] free().
    p_char_array_free(flds, flds_length);

    return 0;
}

int main(int argc, char* argv[])
{
#define fgets_MAX   65536
    FILE *fp;
    char line[fgets_MAX];

    if (argc < 2) {
        printf("Usage %s <testdatafile>\n", argv[0]);
        return -1;
    }

    // File Open
    fp = fopen(argv[1], "r");

    if (fp == NULL) {
        printf("%s Open Failed...\n", argv[1]);
        return -1;
    }

    while((fgets(line, fgets_MAX - 1, fp)) != NULL) {
        trim(line);
        if (*line == '\0')
            continue;
        printf("arg = %s\n", line);
        loop_main(line);
    }

    fclose(fp);
    return 0;
}
